azkaban-common/src/main/java/azkaban/user/UserUtils.java

```diff
@@ -22,4 +33,56 @@ public static boolean hasPermissionforAction(final UserManager userManager, fina
 
     return false;
   }
+
+  /**
+   * Creates a watch thread which listens to specified files' modification and reloads
+   * configurations
+   */
+  public static void setupWatch(final String fileName, Logger log, ParseConfigFile parser) {
+    Runnable runnable = () -> {
```

Comment:
since it's a public API might be used by other UserManager, should we have stricter argument validation? Like file existence check and null pointer check.

Reply:
That is a good point.